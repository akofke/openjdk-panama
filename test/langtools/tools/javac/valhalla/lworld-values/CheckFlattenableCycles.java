/*
 * @test /nodynamiccopyright/
 * @summary Check for cycles through fields declared flattenable.
 *
 * @compile/fail/ref=CheckFlattenableCycles.out -XDrawDiagnostics CheckFlattenableCycles.java
 */

final value class CheckFlattenableCycles {
    class InnerRef {
        CheckFlattenableCycles cfc;
    }
    value final class InnerValue {
        final CheckFlattenableCycles.val cfc = CheckFlattenableCycles.default; // Error.
    }
    final CheckFlattenableCycles cfc = CheckFlattenableCycles.default; // Error.
    final int i = 10;
    final String s = "blah";
    final InnerRef ir = new InnerRef(); // OK.
    final InnerValue.val iv = InnerValue.default; // Error
}
