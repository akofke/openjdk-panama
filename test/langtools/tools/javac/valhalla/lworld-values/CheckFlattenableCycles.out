CheckFlattenableCycles.java:13:42: compiler.err.cyclic.value.type.membership: CheckFlattenableCycles.InnerValue
CheckFlattenableCycles.java:15:34: compiler.err.cyclic.value.type.membership: CheckFlattenableCycles
CheckFlattenableCycles.java:19:26: compiler.err.cyclic.value.type.membership: CheckFlattenableCycles
3 errors
